### ISSUE TITLE
Review all v3 attributes which take Type and add generic versions

### ISSUE THREAD

**Issue by bradwilson** [xunit/xunit#2890]
Similar to how `Collection<T>` support was added in #2629, we should review all attributes which have a Type-based constructor and consider offering generic versions as well.

**Comment by JamesTerwilliger** [comment]
I'm seeing 10 classes with this pattern:
- AssemblyFixtureAttribute
- ClassDataAttribute
- CollectionBehaviorAttribute
- DataDiscovererAttribute
- TestCaseOrdererAttribute
- TestCollectionOrdererAttribute
- TestFrameworkAttribute
- TestFrameworkDiscovererAttribute
- TraitDiscovererAttribute
- XUnitTestCaseDiscovererAttribute

**Comment by bradwilson** [comment]
Yep, all these look like good targets.

**Comment by bradwilson** [comment]
That is correct, we do not support open generics in fixture types (nor is there any plan to, because there is no way for us to know how to close the generic).

You can use a closed generic (e.g., `[assembly: AssemblyFixture(typeof(DatabaseFixture<SomeType>))]`).

**Comment by bradwilson** [comment]
Closing this issue as this was already shipped in `1.0.0`.
